vsutil/__init__.py

```diff
@@ -244,13 +244,16 @@ def depth(clip: vs.VideoNode,
 
     # thanks @Frechdachs for explaining this:
     # 'you need dithering when raising the bitdepth of full range [or] converting between full and limited'
-    # the exception is when converting from 8 to 16 bit full range, (0-255) * 257 -> (0-65535)
-    # dithering is always needed when converting from float to integer precision
-    # dithering is needed when converting from an integer depth greater than 10 to half float, despite the higher bidepth
-    should_dither = (range_in != range
-                     or (clip.format.sample_type != sample_type and (curr_depth + 6) > bitdepth)
-                     or (range_in == Range.FULL and not (curr_depth, bitdepth) == (8, 16))
-                     or curr_depth > bitdepth)
+    # The exception is when converting from 8 to 16 bit full range, (0-255) * 257 -> (0-65535).
+    # Dithering is always needed when converting from float to integer precision.
+    # Dithering is theoretically needed when converting from an integer depth greater than 10 to half float,
+    # despite the higher bitdepth, but zimg currently will not dither for float output.
+    should_dither = (range_in != range  # full->limited OR limited->full
+                     or clip.format.sample_type == vs.FLOAT  # float->int
+                     or (range_in == Range.FULL  # full/int->full/int...
+                         and not (curr_depth, bitdepth) == (8, 16))  # ...but specifically not for 8->16
+                     or curr_depth > bitdepth) and sample_type == vs.INTEGER  # limited/int->limited/int if downsampling
```

Comment:
`sample_type == vs.INTEGER` should be the first condition, because it is required and simple and should most definitely *not* be on the same line as the end of the large disjunction.

Reply:
```py
...
    should_dither = (sample_type == vs.INTEGER
                     and (range_in != range  # full->limited OR limited->full
                          or clip.format.sample_type == vs.FLOAT  # float->int
                          or (range_in == Range.FULL  # full/int->full/int...
                              and (curr_depth, bitdepth) != (8, 16))  # ...but specifically not for 8->16
                          or curr_depth > bitdepth))  # limited/int->limited/int if downsampling
```
This look good?

Reply:
Yes, this is better. I would argue that most of the comments seem rather self-explanatory and don't add to understandability of the code. In fact, is ` limited/int->limited/int if downsampling` even correct, because that condition can also be reached when `(range_in, range) == (Range.FULL, Range.FULL)`?

Reply:
I refactored this into its own function with tests now.